src/Features/CSharp/Portable/ExtractMethod/CSharpSelectionResult.StatementResult.cs

```diff
@@ -89,6 +89,24 @@ public override (ITypeSymbol? returnType, bool returnsByRef) GetReturnTypeInfo(C
                         return default;
                 }
             }
+
+            public override SyntaxNode GetOutermostCallSiteContainerToProcess(CancellationToken cancellationToken)
+            {
+                if (this.IsExtractMethodOnSingleStatement)
+                {
+                    var firstStatement = this.GetFirstStatement();
+                    return firstStatement.GetRequiredParent();
+                }
+
+                if (this.IsExtractMethodOnMultipleStatements)
```

Comment:
>if (this.IsExtractMethodOnMultipleStatements) [](http://example.com/codeflow?start=16&length=45)

personal preference: Contract.ThrowIfFalse and remove the if and Unreachable exception

Reply:
can do in followup.  tnx!